chopper/lib/src/chopper_exception.dart

```diff
@@ -1,24 +1,186 @@
 import 'package:chopper/src/request.dart';
 import 'package:chopper/src/response.dart';
 
-/// {@template ChopperException}
-/// An exception thrown when something goes wrong with Chopper.
-/// {@endtemplate}
-class ChopperException implements Exception {
-  /// {@macro ChopperException}
-  ChopperException(this.message, {this.response, this.request});
+/// Represents the type of error occurred in Chopper.
+enum ChopperExceptionType {
+  connectionTimeout,
+  sendTimeout,
+  receiveTimeout,
+  badResponse,
+  cancel,
+  connectionError,
+  unknown,
+}
+
+extension ChopperExceptionTypeDescription on ChopperExceptionType {
```

Comment:
With enhanced enums in Dart 3, creating extensions like these is no longer necessary. You can simply add a the method `toPrettyDescription` to the `ChopperExceptionType` enum.